app/src/features/surveys/animals/animal-form/edit/EditAnimalPage.tsx

```diff
@@ -0,0 +1,243 @@
+import { APIError } from 'hooks/api/useAxios';
+import { useAnimalPageContext, useDialogContext, useProjectContext, useSurveyContext } from 'hooks/useContext';
+import { useCritterbaseApi } from 'hooks/useCritterbaseApi';
+import { SKIP_CONFIRMATION_DIALOG, useUnsavedChangesDialog } from 'hooks/useUnsavedChangesDialog';
+import { ICreateEditAnimalRequest } from 'interfaces/useCritterApi.interface';
+import { useRef, useState } from 'react';
+import { Prompt, useHistory, useParams } from 'react-router';
+import { Link as RouterLink } from 'react-router-dom';
+
+/**
+ * Returns the page for editing an existing animal (critter) within a Survey
+ *
+ * @return {*}
+ */
+export const EditAnimalPage = () => {
+  const history = useHistory();
+
+  const critterbaseApi = useCritterbaseApi();
+
+  const surveyContext = useSurveyContext();
+  const projectContext = useProjectContext();
+  const dialogContext = useDialogContext();
+  const animalPageContext = useAnimalPageContext();
+
+  const urlParams: Record<string, string | number | undefined> = useParams();
+  const surveyCritterId: number | undefined = Number(urlParams['survey_critter_id']);
+
+  const { locationChangeInterceptor } = useUnsavedChangesDialog();
+
+  const [isSaving, setIsSaving] = useState(false);
+
+  const formikRef = useRef<FormikProps<ICreateEditAnimalRequest>>(null);
+
+  const { projectId, surveyId } = surveyContext;
+
+  // Update the selected animal based on url Params
+  if (surveyCritterId) {
+    animalPageContext.setSelectedAnimalFromSurveyCritterId(surveyCritterId);
+  }
+
+  const critter = animalPageContext.critterDataLoader.data;
+
+  // Loading spinner if the data later hasn't updated to the selected animal yet
+  if (!critter || animalPageContext.selectedAnimal?.critterbase_critter_id !== critter.critter_id) {
+    return <CircularProgress className="pageProgress" size={40} />;
+  }
+
+  const handleCancel = () => {
+    history.push(`/admin/projects/${projectId}/surveys/${surveyId}/animals`);
+  };
+
+  const showCreateErrorDialog = (textDialogProps?: Partial<IErrorDialogProps>) => {
+    dialogContext.setErrorDialog({
+      dialogTitle: EditAnimalI18N.createErrorTitle,
+      dialogText: EditAnimalI18N.createErrorText,
+      onClose: () => {
+        dialogContext.setErrorDialog({ open: false });
+      },
+      onOk: () => {
+        dialogContext.setErrorDialog({ open: false });
+      },
+      ...textDialogProps,
+      open: true
+    });
+  };
+
+  /**
+   * Creates an animal
+   *
+   * @return {*}
+   */
+  const handleSubmit = async (values: ICreateEditAnimalRequest) => {
+    setIsSaving(true);
+
+    try {
+      if (!values.species) {
+        return;
+      }
+
+      const response = await critterbaseApi.critters.updateCritter({
+        critter_id: critter.critter_id,
+        itis_tsn: values.species.tsn,
+        wlh_id: values.wildlife_health_id,
+        animal_id: values.nickname,
+        sex: AnimalSex.UNKNOWN,
```

Comment:
I don't think there is a sex field on the page currently, though maybe there is meant to be and this was a placeholder? 

Probably a @mauberti-bc question.

Reply:
I hardcoded sex to unknown in the critter table so that users add sex as a measurement instead, which gives users more relevant options for their species. We can keep it as is.

Reply:
Do we need to deprecate that sex field in Critterbase then?

We'd want to eventually migrate the existing records to using a measurement then too.

Reply:
Yeah, eventually we will